source/slang/slang-check-decl.cpp

```diff
@@ -1758,6 +1874,348 @@ namespace Slang
+    {
+        // The situation here is that the context of an inheritance
+        // declaration didn't provide an exact match for a required
+        // property. E.g.:
+        //
+        //      interface ICell { property value : int { get; set; } }
+        //      struct MyCell : ICell
+        //      {
+        //          int value;
+        //      }
+        //
+        // It is clear in this case that the `MyCell` type *can*
+        // satisfy the signature required by `ICell`, but it has
+        // no explicit `property` declaration, and instead just
+        // a field with the right name and type.
+        //
+        // The approach in this function will be to construct a
+        // synthesized `preoperty` along the lines of:
+        //
+        //      struct MyCounter ...
+        //      {
+        //          ...
+        //          property value_synthesized : int
+        //          {
+        //              get { return this.value; }
+        //              set(newValue) { this.value = newValue; }
+        //          }
+        //      }
+        //
+        // That is, we construct a `property` with the correct type
+        // and with an accessor for each requirement, where the accesors
+        // all try to read or write `this.value`.
+        //
+        // If those synthesized accessors all type-check, then we can
+        // say that the type must satisfy the requirement structurally,
+        // even if there isn't an exact signature match. More
+        // importantly, the `property` we just synthesized can be
+        // used as a witness to the fact that the requirement is
+        // satisfied.
+        //
+        // The big-picture flow of the logic here is similar to
+        // `trySynthesizeMethodRequirementWitness()` above, and we
+        // will not comment this code as exhaustively, under the
+        // assumption that readers of the code don't benefit from
+        // having the exact same information stated twice.
+
+        // With the introduction out of the way, let's get started
+        // constructing a synthesized `PropertyDecl`.
+        //
+        auto synPropertyDecl = m_astBuilder->create<PropertyDecl>();
+
+        // For now our synthesized property will use the name and source
+        // location of the requirement we are trying to satisfy.
+        //
+        // TODO: as it stands right now our syntesized property and its
+        // accesors will get mangled names, which we don't actually want.
+        // Leaving out the name here doesn't help matters, becaues then
+        // *all* synthesized members on a given type would share the same
+        // mangled name.
+        //
+        synPropertyDecl->nameAndLoc = requiredMemberDeclRef.getDecl()->nameAndLoc;
+
+        // The type of our synthesized property will be the expected type
+        // of the interface requirement.
+        //
+        // TODO: This logic can/will run into problems if the type is,
```

Comment:
Is there an assumption that synthesized code will not get type checked again? This seems to be one place that you'll want to check synthesized code for simplicity.

Reply:
Yes, this logic runs late enough that we shouldn’t have problems with things getting re-checked.